crates/sui-adapter/src/adapter.rs

```diff
@@ -337,7 +347,13 @@ pub fn publish<
             CompiledModule::deserialize(b)
                 .map_err(|e| e.finish(move_binary_format::errors::Location::Undefined))
         })
-        .collect::<move_binary_format::errors::VMResult<Vec<CompiledModule>>>()?;
+        .collect::<move_binary_format::errors::VMResult<Vec<CompiledModule>>>()
+        .map_err(|e| {
+            ExecutionError::new_with_source(
+                sui_types::messages::ExecutionFailureStatus::VMInvariantViolation,
+                e,
+            )
+        })?;
```

Comment:
This change is what's needed to get rid of the previous `NullResolver`: in order to invoke `convert_vm_error`, we need a handle on `vm`. But in this `publish` function, there is no `vm` handle (and no need for it based on the runtime).

I dug into how this could possibly fail and what error it produces. After looking at `CompiledModule::deserialize(b)` and  `e.finish`, it was clear that this can only fail with [`VERIFIER_INVARIANT_VIOLATION`](https://sourcegraph.com/github.com/move-language/move/-/blob/language/move-binary-format/src/deserializer.rs?L54-56&subtree=true), code `2016` which ends up being converted to `sui_types::messages::ExecutionFailureStatus::VMInvariantViolation` via:

https://sourcegraph.com/github.com/move-language/move/-/blob/language/move-core/types/src/vm_status.rs?L705-709&subtree=true

https://sourcegraph.com/github.com/move-language/move/-/blob/language/move-core/types/src/vm_status.rs?L29-32&subtree=true

https://sourcegraph.com/github.com/MystenLabs/sui/-/blob/crates/sui-types/src/error.rs?L759&subtree=true

Reply:
Sorry, I'm not sure that I follow. The error here could be a result from any invalid series of bytes attempting to be published. So why are we mapping the error to VM Invariant violation? 

e.g. if I pass in `publish(.., vec![vec![0]], ..)` I should get a deserialization error. Won't this code here just switch that into an invariant violation error?

Reply:
...you're right. I overlooked this somehow and this is dumb.

> [...] shuffle around publish so we have the VM instance?

after passing `vm` in `publish`, we can just call out to `convert_vm_error` here, so it's fixed